cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ImpersonationUserResolver.java

```diff
@@ -46,11 +46,11 @@
   /**
    * Get Impersonated user at namespace level. If it is empty, returns the default principal.
    *
-   * @param programId NamespaceId
    * @return configured principal.
    */
-  String getPrincipal(Id.Program programId) {
-    String namespaceUserSetting = getNamespaceConfig(programId.getNamespace()).getPrincipal();
+  @Nullable
```

Comment:
It could be the case that the cconf doesn't have the property defined.
Instead, I'll update the calling code to only call this if security is enabled.